addresshelper/helper.go

```diff
@@ -17,25 +19,23 @@ func PkScript(address string) ([]byte, error) {
 }
 
 func DecodeForNetwork(address string, params *chaincfg.Params) (dcrutil.Address, error) {
-	addr, err := dcrutil.DecodeAddress(address)
+	addr, err := dcrutil.DecodeAddress(address, params)
 	if err != nil {
 		return nil, err
 	}
-	if !addr.IsForNet(params) {
-		return nil, fmt.Errorf("address %s is not intended for use on %s", address, params.Name)
-	}
+
 	return addr, nil
 }
 
 func PkScriptAddresses(params *chaincfg.Params, pkScript []byte) ([]string, error) {
-	_, addresses, _, err := txscript.ExtractPkScriptAddrs(txscript.DefaultScriptVersion, pkScript, params)
+	_, addresses, _, err := txscript.ExtractPkScriptAddrs(scriptVersion, pkScript, params)
```

Comment:
Why are we using a custom defined script version?

Reply:
`txscript.DefaultScriptVersion` was removed in v2 but `txscript.ExtractPkScriptAddrs` only attempts to verify version 0 scripts.